### PR TITLE
Add Cosmos GetByQueryAsync function overload with QueryDefinition

### PR DESCRIPTION
**Pull request by afacanerman**
Added a function overload for GetByQueryAsync allowing users to put their query parameters

### CODE REVIEW

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs [hunk 1]
```diff
@@ -166,7 +192,7 @@ static async Task ServiceExampleAsync(IExampleService service)
 
             // Creating...
             Console.WriteLine("[Person] Service creating...");
-            _ = await service.AddPeopleAsync(new[] { jamesBond, adeleGoldberg });
+            _ = await service.AddPeopleAsync(new[] {jamesBond, adeleGoldberg});
```
Comment (IEvangelist) [inline Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs:195]:
```suggestion
            _ = await service.AddPeopleAsync(new[] { jamesBond, adeleGoldberg });
```

Undoing some of the undesired formatting changes.

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs [hunk 2]
```diff
@@ -116,11 +127,12 @@ static async Task RawRepositoryExampleAsync(IRepository<Widget> repository)
 
             // Creating...
             Console.WriteLine("[Widget] Repository creating...");
-            _ = await repository.CreateAsync(new[] { widget1, widget2 });
+            _ = await repository.CreateAsync(new[] {widget1, widget2});
```
Comment (IEvangelist) [inline Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs:130]:
```suggestion
            _ = await repository.CreateAsync(new[] { widget1, widget2 });
```

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs [hunk 3]
```diff
@@ -68,7 +69,7 @@ static async Task RawRepositoryExampleAsync(IRepository<Person> repository)
 
             // Creating...
             Console.WriteLine("[Person] Repository creating...");
-            _ = await repository.CreateAsync(new[] { maryShaw, calvinWeatherfield });
+            _ = await repository.CreateAsync(new[] {maryShaw, calvinWeatherfield});
```
Comment (IEvangelist) [inline Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs:72]:
```suggestion
            _ = await repository.CreateAsync(new[] { maryShaw, calvinWeatherfield });
```

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs [hunk 4]
```diff
@@ -43,13 +44,13 @@ static async Task Main(string[] args)
...
-                                options.ContainerPerItemType = true;
-                            })
-                            .AddSingleton<IExampleService, ExampleService>());
+                        {
+                            options.ContainerId = "people-store";
+                            options.DatabaseId = "samples";
+                            options.OptimizeBandwidth = true;
+                            options.ContainerPerItemType = true;
+                        })
+                        .AddSingleton<IExampleService, ExampleService>());
```
Comment (IEvangelist) [inline Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs:53]:
```suggestion
                            {
                                options.ContainerId = "people-store";
                                options.DatabaseId = "samples";
                                options.OptimizeBandwidth = true;
                                options.ContainerPerItemType = true;
                            })
                            .AddSingleton<IExampleService, ExampleService>());
```

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs [hunk 5]
```diff
@@ -103,6 +104,16 @@ static async Task RawRepositoryExampleAsync(IRepository<Person> repository)
...
         {
+            async Task VerifyUpdates()
+            {
+                // Read again / verify updates
+                IEnumerable<Widget> validWidgets = await repository.GetAsync(p => p.Name != null);
+                foreach (Widget widget in validWidgets)
+                {
+                    Console.WriteLine($"[Widget] Updated: {widget}");
+                }
+            }
```
Comment (IEvangelist) [inline Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs:115]:
```suggestion
            static async Task VerifyUpdates(IRepository<Widget> repo)
            {
                // Read again / verify updates
                IEnumerable<Widget> validWidgets = await repo.GetAsync(p => p.Name != null);
                foreach (Widget widget in validWidgets)
                {
                    Console.WriteLine($"[Widget] Updated: {widget}");
                }
            }
```

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs [hunk 6]
```diff
@@ -133,12 +145,26 @@ static async Task RawRepositoryExampleAsync(IRepository<Widget> repository)
...
+
+            // Updating...
+            Console.WriteLine("[Widget] Repository updating...");
+            contraption.CreatedOrUpdatedOn = contraption.CreatedOrUpdatedOn.AddDays(1);
+            telescope.CreatedOrUpdatedOn = telescope.CreatedOrUpdatedOn.AddDays(1);
+
+            _ = repository.UpdateAsync(contraption);
+            _ = repository.UpdateAsync(telescope);
+
+            await VerifyUpdates();
```
Comment (IEvangelist) [inline Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs:167]:
```suggestion
            await VerifyUpdates(repository);
```

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs [hunk 7]
```diff
@@ -133,12 +145,26 @@ static async Task RawRepositoryExampleAsync(IRepository<Widget> repository)
             _ = repository.UpdateAsync(contraption);
             _ = repository.UpdateAsync(telescope);
 
-            // Read again / verify updates
-            IEnumerable<Widget> validWidgets = await repository.GetAsync(p => p.Name != null);
-            foreach (Widget widget in validWidgets)
-            {
-                Console.WriteLine($"[Widget] Updated: {widget}");
-            }
+            await VerifyUpdates();
```
Comment (IEvangelist) [inline Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs:148]:
```suggestion
            await VerifyUpdates(repository);
```

**Review by IEvangelist — CHANGES_REQUESTED** [review]
I'm making a few minor style changes, overall I really like this - thank you for these and I'm looking forward to your reply. Thanks

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs (hunk 4 above)
Comment (afacanerman) [reply Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs]: Sorry, these were my IDE's default. I rolled them back

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs (hunk 3 above)
Comment (afacanerman) [reply Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs]: done

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs (hunk 5 above)
Comment (afacanerman) [reply Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs]: done

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs (hunk 2 above)
Comment (afacanerman) [reply Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs]: done

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs (hunk 7 above)
Comment (afacanerman) [reply Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs]: done

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs (hunk 6 above)
Comment (afacanerman) [reply Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs]: done

File: Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs (hunk 1 above)
Comment (afacanerman) [reply Microsoft.Azure.CosmosRepository.Samples/ServiceTier/Program.cs]: done

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

**Comment by IEvangelist** [comment]
@all-contributors please add @afacanerman for code and examples

**Pull request merged**